### PR TITLE
Set Env Var ASPNET_CONTENTROOT to /azure-functions-host - to limit filewatchers

### PR DESCRIPTION
**Pull request by CooperLink**
Related to recent issue https://github.com/Azure/azure-functions-docker/issues/628

Update all images to add the ASPNET_CONTENTROOT variable to the environment.  This PR incorporates the changes introduced by hotfix releases 
https://github.com/Azure/azure-functions-docker/releases/tag/4.1.3.1
https://github.com/Azure/azure-functions-docker/releases/tag/3.4.2.1

A recent incident found that the Azure-Functions-Host spawned a very high number of inotify filewatchers on startup.  The cause was found to be a WebHostBuilder object in the Functions-Host code. The object [inherits ASPNET_CONTENTROOT as its context](https://github.com/aspnet/MetaPackages/blob/master/src/Microsoft.AspNetCore/WebHost.cs#L152).  Without that environment variable set, the object defaults to the `current directory` which is currently `/` for the linux images. 

This PR provides `ASPNET_CONTENTROOT = /azure-functions-host` to the WebHostBuilder which decreases the number of inotify filewatchers spawned on startup significantly.

### PR information

- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/development-docs/cleaning-up-commits.md).
- [x] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### CODE REVIEW

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template
Comment (pragnagopa) [inline host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template]: change this to all caps - https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-6.0#environment

**Comment by pragnagopa** [comment]
@VpOfEngineering - please wait for this PR to be merged before starting release

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template
Comment (balag0) [inline host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:13]: I might have missed this in the emails, but why do we need HOST_VERSION?

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template
Comment (balag0) [inline host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: From https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/web-host?view=aspnetcore-3.1#content-root, this should default to the folder where the app assembly resides.
Since we are not setting this currently , in these images shouldnt that already be /azure-functions-host ?

Comment (pragnagopa) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: very likely this is a bug - needs follow up with .net team - file watchers count drastically went down after setting this explicitly. @balag0  very like need to set this in mesh images as well until we figure this out in how to set this in functions host code.

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template
Comment (pragnagopa) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:13]: can you expand? this is an existing setting

Comment (balag0) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:13]: sorry, i was using the wrong diff view :)

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template
Comment (balag0) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: Just a quick guess (but would be good to validate), based on https://github.com/aspnet/MetaPackages/blob/master/src/Microsoft.AspNetCore/WebHost.cs#L152
if the env. variable is not set, it defaults to current directory.
In the dockerfiles there is no WORKDIR is set. so the current directory at that point could be /

If we have a way to repro/validate, can we add 
WORKDIR /azure-functions-host
to the dockerfiles and not set ASPNETCORE_CONTENTROOT explicitly
and see if that fixes the issue?

Comment (pragnagopa) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: good idea. @CooperLink - it is worth it to setup a repro for this on private stamp - host image without this change and Storage app setting - mount Azure Files with large folder structure.

Comment (CooperLink) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: After pushing hotfix changes to UD6 I will be setting up a private stamp repro for this to test the above. Thank you both for your comments

Comment (pragnagopa) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: @balag0  - @CooperLink  brought up a good point - would setting WorkingDir affect applications? I remember we considered this inthe past and decided was a breaking change - but OK to do this in V4 while it was in preview.

Comment (balag0) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: Yes, had a chat with Cooper just now as well. Yes agree that using the appsetting is the safer and better fix. Setting the WORKDIR was just an experiment to make sure we understand the behavior (in case there was a bug in the framework) :)

Also here is an approach to set this in the code (but dockerfiles is still better fix for I think)
https://stackoverflow.com/questions/59033413/c-sharp-aspnetcore-set-content-root-path

**Review by balag0 — APPROVED** [review]

**Comment by pragnagopa** [comment]
@CooperLink - please update the PR description before merging.

File: host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template
Comment (pragnagopa) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: @balag0 - is this change needed in mesh images?

Comment (balag0) [reply host/3.0/buster/amd64/dotnet/dotnet-inproc/dotnet-composite.template:14]: Potentially since the setup is the same, but given we dont have any CRIs and the underlying issue here was exposed only after a config update on workers, I think its not urgent and we can wait until these images are rolled out and tested in case we identify any other scenarios to consider.

**Pull request merged**
